File: src/main/java/set/params/EnumNumber.java
package set.params;

public enum EnumNumber implements Parameter{
	
	NUMBER_1("1"),
	NUMBER_2("2"),
	NUMBER_3("3");
	
	private final String label;
	
	EnumNumber(String label) {
		this.label = label;
	}
	
	public String getLabel() {
		return label;
	}
	
}

File: src/main/frontend/src/game/model/card_params/EnumNumber.js
import Parameter from "/game/model/card_params/Parameter.js"

export default class EnumNumber extends Parameter {
	
	/**
	 * @param {String} label 
	 * @param {Number} ID
	 */
	constructor(label, ID) {
		super(label, ID);
	}

	//STATICS ************************************

	static NUMBER_1 = new EnumNumber("1", 0);
	static NUMBER_2 = new EnumNumber("2", 1);
	static NUMBER_3 = new EnumNumber("3", 2);

	/**
	 * @type {Array<EnumNumber>}
	 */
	static #constants = [this.NUMBER_1, this.NUMBER_2, this.NUMBER_3];

	/**
	 * @param {Number} ID
	 * @returns {EnumNumber}
	 */
	static valueOf(ID) {
		return this.#constants[ID];
	}

}